tests/integration/consumer/test_cooperative_rebalance_1.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limit
+
+import pytest
+import time
+from uuid import uuid1
+
+
+def test_cooperative_rebalance_1(kafka_cluster):
+    """
+    Test cooperative-sticky assignor and various aspects
+    of the incremental rebalancing API.
+    """
+
+    consumer_conf = {'group.id': str(uuid1()),
+                     'partition.assignment.strategy': 'cooperative-sticky',
+                     'enable.auto.commit': 'false',
+                     'auto.offset.reset': 'earliest',
+                     'heartbeat.interval.ms': '2000',
+                     'session.timeout.ms': '6000', # minimum allowed by broker
+                     'max.poll.interval.ms': '6500'}
+
+    assign_count = 0
+    def on_assign(consumer, partitions):
+        nonlocal assign_count
+        assign_count += 1
+        assert 1 == len(partitions)
+
+    revoke_count = 0
+    def on_revoke(consumer, partitions):
+        nonlocal revoke_count
+        revoke_count += 1
+
+    lost_count = 0
+    def on_lost(consumer, partitions):
+        nonlocal lost_count
+        lost_count += 1
+
+    topic1 = kafka_cluster.create_topic("topic1")
+    topic2 = kafka_cluster.create_topic("topic2")
+
+    consumer = kafka_cluster.consumer(consumer_conf)
+
+    kafka_cluster.seed_topic(topic1, value_source=[b'a'])
+
+    consumer.subscribe([topic1], on_assign=on_assign, on_revoke=on_revoke, on_lost=on_lost)
+    msg = consumer.poll(10)
+    assert msg != None
+    assert msg.value() == b'a'
+
+    # Subscribe to a second one partition topic, the second assign
+    # call should be incremental (checked in the handler).
+    consumer.subscribe([topic1, topic2], on_assign=on_assign, on_revoke=on_revoke, on_lost=on_lost)
+
+    kafka_cluster.seed_topic(topic2, value_source=[b'b'])
+    msg2 = consumer.poll(10)
+    assert msg2 != None
+    assert msg2.value() == b'b'
+
+    assert 2 == assign_count
+    assert 0 == lost_count
+    assert 0 == revoke_count
+
+    msg3 = consumer.poll(1)
+    assert msg3 == None
+
+    # Exceed MaxPollIntervalMs => lost partitions.
+    time.sleep(8)
+    
+    consumer.close()
+
+    assert 2 == assign_count
+    assert 1 == lost_count
```

Comment:
Do we really know that the lost is from maxpoll timeout and not from the close?
Better specifically poll after sleep to make sure the lost is seen, then do close.